cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -235,12 +279,14 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
     MacroParser macroParser = new MacroParser(trackingMacroEvaluator);
 
     for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
-      if (pluginEntry.getValue().isMacroSupported()) {
+      if (pluginEntry.getValue() != null && pluginEntry.getValue().isMacroSupported()) {
```

Comment:
I think this part was changed with Sagar's fix. If you rebase you'll pick it up.

Should be looping over the plugin properties instead of the plugin property fields